scheduler/src/cook/mesos/api.clj

```diff
@@ -1144,13 +1152,149 @@
      :exists? (partial retrieve-jobs conn)
      :handle-ok (fn [ctx] (render-jobs-for-response-deprecated conn framework-id ctx))}))
 
+(defn valid-name-filter?
+  "Returns true if the provided name filter is either nil or satisfies the JobNameListFilter schema"
+  [name]
+  (or (nil? name)
+      (nil? (s/check JobNameListFilter name))))
+
+(defn normalize-list-states
+  "Given a set, states, returns a new set that only contains one of the
+  'terminal' states of completed, success, and failed. We take completed
+  to mean both success and failed."
+  [states]
+  (if (contains? states "completed")
+    (set/difference states util/instance-states)
+    (if (set/superset? states util/instance-states)
+      (-> states (set/difference util/instance-states) (conj "completed"))
+      states)))
+
+(defn parse-time
+  "Parses the provided string as a DateTime"
+  [s]
+  (or (tf/parse s)
+      (tc/from-long (Long/parseLong s))))
+
+(defn- parse-int-default
+  "Parses the provided string as an integer"
+  [s d]
+  (if (nil? s)
+    d
+    (Integer/parseInt s)))
+
+(defn name-filter-str->name-filter-pattern
+  "Generates a regex pattern corresponding to a user-provided name filter string"
+  [name]
+  (re-pattern (-> name
+                  (str/replace #"\." "\\\\.")
+                  (str/replace #"\*+" ".*"))))
+
+(defn name-filter-str->name-filter-fn
+  "Returns a name-filtering function (or nil) given a user-provided name filter string"
+  [name]
+  (when name
+    (let [pattern (name-filter-str->name-filter-pattern name)]
+      #(re-matches pattern %))))
+
+(defn job-list-request-malformed?
```

Comment:
Missing docstring

Reply:
Are `start` and `end` required parameters?

Reply:
Done